modules/standard/Path.chpl

```diff
@@ -421,4 +421,69 @@ proc file.realPath(): string throws {
   return result;
   }
 
+  /* Collapse paths such as `foo//bar`, `foo/bar/`, `foo/./bar`, and 
+   `foo/baz/../bar` into `foo/bar`.  Warning: may alter meaning of paths
+   containing symbolic links.  Similar to :proc:`normCase`, on Windows will replace
+   forward slashes.
+
+   :arg name: a potential path to collapse, possibly destroying the meaning of the
+             path if symbolic links were included.
+   :type name: `string`
+
+   :return: the collapsed version of `name`
+   :rtype: `string`
+  */
+
+  proc normPath(name: string): string {
+
+  if name.isEmptyString() {
+      return name;
+  }
+
+  //stores the result string
+  var result = name;
+
+  if result.find("//") then
+    result = result.replace("//", "/", -1);
+
+  if result.find("/./") then
+    result = result.replace("/./", "/", -1);
```

Comment:
This condition will miss paths ending with `.`, e.g. `foo/bar/.`.  It makes sense to handle the removal like this, so you'll want to perform that check on the end of the string.

Reply:
You try to avoid creating that situation for yourself, but a caller could still send it to you